packages/repository/src/relations/relation.types.ts

```diff
@@ -68,6 +68,42 @@ export interface HasManyDefinition extends RelationDefinitionBase {
   keyTo?: string;
 }
 
+export interface HasManyThroughDefinition extends RelationDefinitionBase {
+  type: RelationType.hasMany;
```

Comment:
> I think we should use a new relation type here, e.g. `RelationType.hasAndBelongsToMany`?

☝️ that does not make sense, this PR is adding "has many through", not "has and belongs to many".

Having said that, is it a good idea to keep using `hasMany` type or should we introduce a new relation type `RelationType.hasManyThrough`?

/cc @strongloop/loopback-maintainers

Reply:
I think if we shouldn't add `RelationType.hasManyThrough`. If it's added, what's the point of hasManyThrough being part of hasMany?

Reply:
So far, we are using `type` to discriminate between different interfaces inheriting from `RelationDefinitionBase`. See https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

If we want to keep using `hasMany` value for has-many-through relations, then I am fine with that, but let's capture that in the type system too.

We can either get rid of `HasManyThroughDefinition` completely and move all members to `HasManyDefinition`, or else modify `HasManyThroughDefinition` to inherit from `HasManyDefinition`.

I think we need to fix `@hasMany` decorator too, to ensure it describes  "through" fields too. This is important to allow TypeScript to verify metadata related to "through" part and allow IDEs to offer auto-completion.

It's possible you already modified `@hasMany` decorator, I didn't checked - the patch is too large to navigate around.

Reply:
So far, we are using `type` to discriminate between different interfaces inheriting from `RelationDefinitionBase`. See https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

If we want to keep using `hasMany` value for has-many-through relations, then I am fine with that, but let's capture that in the type system too.

We can either get rid of `HasManyThroughDefinition` completely and move all members to `HasManyDefinition`, or else modify `HasManyThroughDefinition` to inherit from `HasManyDefinition`.

I think we need to fix `@hasMany` decorator too, to ensure it describes  "through" fields too. This is important to allow TypeScript to verify metadata related to "through" part and allow IDEs to offer auto-completion.

It's possible you already modified `@hasMany` decorator, I didn't checked - the patch is too large to navigate around.

Reply:
> hasAndBelongsToMany

I thought this will be added in a separate pull request? `hasAndBelongsToMany` is dependant on `hasManyThrough`.

https://github.com/strongloop/loopback-next/issues/2308